lib/rack/query_parser.rb

```diff
@@ -118,17 +101,11 @@ def parse_query_pairs(qs, separator = nil, &unescaper)
     def parse_nested_query(qs, separator = nil)
       params = make_params
 
-      unless qs.nil? || qs.empty?
-        check_query_string(qs, separator).split(separator ? (COMMON_SEP[separator] || /[#{separator}] */n) : DEFAULT_SEP).each do |p|
-          k, v = p.split('=', 2).map! { |s| unescape(s) }
-
-          _normalize_params(params, k, v, 0)
-        end
+      each_query_pair(qs, separator) do |k, v|
```

Comment:
I'd like to avoid abbreviations.
```suggestion
      each_query_pair(query_separator, separator) do |key, value|
```

Abbreviations make it harder to understand code.

Reply:
`qs` is for `query_string`, not `query_separator`. While I don't object to using longer variable names, I don't see the need, and it wouldn't necessarily be consistent. In this method, we are calling `make_params` and `_normalize_params` methods, and all existing variables other than `separator` are already abbreviated and, IMO, easy to understand.

Abbreviating variables makes it easier to read code, as long as the abbreviated variable names are understandable. It's common to use `k` and `v` instead of `key` and `value`, just like it is common to use `i` instead of `index`.  Unnecessary verbosity leads to increased fatigue when reading the code.

Reply:
I basically disagree, and the fact that I didn't even know what the expansion was proves my point haha.

Reply:
None of those three variables are newly introduced here; if we did want to change any/all of them it feels like that would be better done all at once, and probably not in a commit that's also making other changes?